Docs/gh 6 read api spec

alexandru-pena opened the pull request:
Added Programme read API

alexandru-pena committed:
added programme and programme offer read api

alexandru-pena committed:
Update programme.md

alexandru-pena committed:
Update programme.md

alexandru-pena committed:
Update programme.md

alexandru-pena committed:
Update programme.md

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -0,0 +1,253 @@
+A programme is composed by a set of _offers_. 
+
+## Properties
+   * Name
+      -  mandatory: the programme's name
```
```suggestion
   * Name: the programme's name
      -  mandatory
```

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -0,0 +1,253 @@
+        "acronym": "LEIC",
+        "termSize": 4
+    },
+    "entities": [
+        {
+            "class": [ "offer" ],
+            "title": "LS Offer",
+            "rel": [ "/rel/programmeOffer" ],
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/LEIC/offers/1"}
+            ]
+        },
+        {
+            "class": [ "offer" ],
+            "title": "AED Offer",
+            "rel": [ "/rel/programmeOffer" ],
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/LEIC/offers/2"}
+            ]
+        },
+        {
+            "class": [ "offer" ],
+            "title": "POO Offer",
+            "rel": [ "/rel/programmeOffer" ],
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/LEIC/offers/3"}
+            ]
+        }
+    ],
```
Objects in the `entities` property can have a `properties` property. It could be interesting to have the terms in which the course is offered

JorgePBrown commented on docs/api/programme.md:
````diff
@@ -0,0 +1,253 @@
+```json
+{
+    "class": [ "collection", "programme" ],
+    "entities": [
+        {
+            "class": [ "programme" ],
+            "rel": [ "item" ],      
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/LEIC" }
+            ]
+        },
+        {
+            "class": [ "Programme" ],
+            "rel": [ "item" ],
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/MEIC" }
+            ]
+        }
+    ],
````
Maybe here they could have term size in the `properties` aswell

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -0,0 +1,253 @@
+   A ProgrammeOffer may have a set of pre-conditions, for example the curricularTerm set where the offer is available or the curricularUnits that precede it.
+
+# Properties
+   * Acronym
+      - mandatory: the curricular unit acronym offered
```
```suggestion
   * Acronym: the curricular unit acronym offered
      - mandatory
```

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -0,0 +1,253 @@
+    "entities": [
+        {
+            "class": [ "course" ],
+            "rel": [ "/rel/course" ],
```
Should the relation be `/rel/precedent`?

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -0,0 +1,253 @@
+        },
+        {
+            "class": [ "course" ],
+            "rel": [ "/rel/course" ],
```
Same here

alexandru-pena committed:
Update programme.md

Updating READ API according to the last meeting decisions.

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -208,45 +219,32 @@ A collection of the possible programmes.
+        "rel": [ "/rel/course/" ],
+        "links": [
+            {"rel": [ "self" ], "href": "/v0/courses/1"}
+        ]
```
I think you removed the { } on accident

```json
"entities": [
  {
    "class": [ "course" ],
    "rel": [ "/rel/course" ],
    "links": [
      {"rel": [ "self" ], "href": "/v0/courses/1"}
    ]
  }
]
```

alexandru-pena replied:
Indeed

jmchamica commented on docs/api/programme.md:
```diff
@@ -0,0 +1,251 @@
+            "properties": {
+                "id": 1,
+                "acronym": "LEIC",
+                "termSize": 6
```
Term size is a detail that should be present, in my opinion, in the Program's full representation. The program's representation as a collection item should only contain essencial information such as the ID and Acronym.

jmchamica commented on docs/api/programme.md:
```diff
@@ -0,0 +1,251 @@
+            "href": "/v0/programmes/",
+            "type": "application/json",
+            "fields": [
+                { "name": "ProgrammeAcr", "type": "text"},
```
Fields should be in camelCase.

jmchamica commented on docs/api/programme.md:
```diff
@@ -0,0 +1,251 @@
+        "termNumber": 2,
+        "optional": "true"
+    },
+    "entities": [
```
Missing {}

jmchamica commented on docs/api/programme.md:
```diff
@@ -0,0 +1,251 @@
+                "TermNumber": 3
+            },
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/LEIC/offers/1"}
```
Is it /v0/programmes/LEIC, or /v0/programmes/1, after all?

alexandru-pena replied:
/v0/programmes/1

Alex-Pena-Ok replied:
/v0/programmes/1

jmchamica commented on docs/api/programme.md:
```diff
@@ -0,0 +1,251 @@
+            "title": "POO Offer",
+            "rel": [ "/rel/programmeOffer" ],
+            "properties": {
+                "CourseId": 4,
```
As an embed, I think the Term Number shouldn't be here.
Also, properties' object should have the offer's ID.

alexandru-pena committed:
Update programme.md

jmchamica committed:
Updates the read API. Course ID is now the Course identifier.

Couse acronym is now an optional property which will, in most cases, be
present in the representation.
A number is used in the path to identify a specific course resource,
promoting cool URIs.

jmchamica committed:
Merge branch 'docs/gh-6-read-api-spec' of github.com:i-on-project/core into docs/gh-6-read-api-spec

jmchamica committed:
Corrects course.md; use /rel/calendar instead of /rel/events.

jmchamica committed:
Corrections.

Commit:
Updated events.md to accomodate new entity referencing

JorgePBrown commented on docs/api/class-sections.md:
```diff
@@ -6,8 +6,12 @@ The `class section` resource has two possible representations: a detailed (or fu
 All properties which are not assigned with the `mandatory` label, are optional, they may not be included in the representation.
 
-* `course`: the course's unique acronym
+* `courseId`: the course's unique identifier
   - mandatory
+  - type: **number**
+  - e.g. 1
+
+* `courseAcr`: the course's unique acronym
   - type: **text**
   - e.g. "WAD"
```
Shouldn't the properties not have relationship information, so no `courseId`, `courseAcr` and `class`? I think the only property the `ClassSection` should have is its Id. And why does it need to have a relation to `Course` if it already has one to `Class`?

JorgePBrown commented on docs/api/classes.md:
```diff
@@ -55,7 +60,7 @@ Since the number of class sections is thought to be very reduced, all of them wi
       "title": "Class Section of Course WAD at Calendar Term 1920v",
       "links": [
-      { "rel" : [ "self" ], "href": "/v0/courses/wad/classes/1920v/1d" },
+      { "rel" : [ "self" ], "href": "/v0/courses/1/classes/1920v/1d" },
```
Isn't the `Class` now referenced using an Id instead of a `CalendarTerm`?

jmchamica replied:
API interactions are independent from the physical model. User interaction doesn't have to change even if the underlying database changes.

JorgePBrown commented on docs/api/classes.md:
```diff
@@ -7,11 +7,15 @@ The `class` resource has three possible representations: a detailed (or full) re
-* `course`: the course's unique acronym
-  - mandatory
+* `courseAcr`: the course's unique acronym
   - type: **text**
   - e.g. "WAD"
 
+* `courseId`: the course's unique identifier
+  - mandatory
+  - type: **number**
+  - e.g. 1
+
 * `calendar term`: period when the academic `event`s of the `course` will occur
   - mandatory
   - type: **text**
```
Same thing as the `ClassSection` relation comment

jmchamica replied:
API interactions are independent from the physical model. User interaction doesn't have to change even if the underlying database does so.

Commit:
Updated events.md to fix some errors that were missed

JorgePBrown commented on docs/api/programme.md:
```diff
@@ -0,0 +1,251 @@
+                "TermNumber": 3
+            },
+            "links" : [
+                { "rel": [ "self" ], "href": "/v0/programmes/1/offers/1"}
```
Maybe it could have a link to the Course aswell

jmchamica committed:
Updates the Home Document spec.

Commit:
Updated: Class representation, empty parameters, uid data type

jmchamica committed:
hrefTemplate of JSON Home uses RFC6570's notation.
